### ISSUE TITLE
No seed for sampling methods

### ISSUE THREAD

**Issue by echo66** [jmschrei/pomegranate#499]
I noticed that https://github.com/jmschrei/pomegranate/blob/df3a46cbff17d1f6c91fdc0e8f7f782f8f723b0c/pomegranate/bayes.pyx#L135 doesn't allow any seed to be passed in order to make the sampling procedure deterministic. My suggestion is doing the same way scikit learn does: add the `random_state` parameter in the method in the method signature and use `sklearn.utils.check_random_state` in order to check if the value is valid and, also, to get a random numbers generator from it.

**Comment by jmschrei** [comment]
Good call. I'll go ahead and try adding in random seed support for sampling now based on an optional argument passed in. This is distinct from training using a random seed, which I've yet to support fully.

Also, I like your avatar picture thing. :)
